appveyor.yml

```diff
@@ -19,4 +24,14 @@ deploy:
     secure: q9CTIytwzn07gcypWBXc1JNw5V1+kFX54YosfZYxGAKchBoCCr4Gpduso4ntoKw2
   on:
     branch: master
-    appveyor_repo_tag: true
\ No newline at end of file
+    appveyor_repo_tag: true
+on_success:
+  - ps: |
+        Write-Host "APPVEYOR_REPO_BRANCH $($env:APPVEYOR_REPO_BRANCH)"
+        Write-Host "APPVEYOR_PULL_REQUEST_NUMBER $($env:APPVEYOR_PULL_REQUEST_NUMBER)"
+        Write-Host "APPVEYOR_PULL_REQUEST_HEAD_REPO_BRANCH  $($env:APPVEYOR_PULL_REQUEST_HEAD_REPO_BRANCH )"
+
+        if (!$env:APPVEYOR_PULL_REQUEST_NUMBER) 
```

Comment:
You should build docs on every branch but deploy only if it's master

Reply:
This is really tricky: as noted in the appveyor documentation **secure variables are not decoded during Pull Request builds**, which means, that only in one of the 2 builds you can push to git with the decrypted token. (see here https://www.appveyor.com/docs/build-configuration/#secure-variables)

Unfortunately, it is exactly the build, where the target branch is NOT specified in any appveyor environment variables (as you can see as I write them to the console a few lines above).

Now I don't know how we could decide whether or not to push the created docu. Maybe you have an idea?

Reply:
You can't deploy on pull request builds anyway, because you only want to do this when the pull request is merged to master. Not when it's created.
Can you just check if APPVEYOR_REPO_BRANCH is master? I think then it will deploy only after something is merged or pushed to master.

Reply:
Done. For testing purposes I want to deploy when the PR is merged to the `docu` branch. if that works as expected, then I'm going to change it to `master`.